homeassistant/components/lutron/switch.py

```diff
@@ -50,3 +61,48 @@ def update(self):
         """Call when forcing a refresh of the device."""
         if self._prev_state is None:
             self._prev_state = self._lutron_device.level > 0
+
+
+class LutronLed(LutronDevice, SwitchDevice):
+    """Representation of a Lutron Keypad LED."""
+
+    def __init__(self, area_name, keypad_name, scene_device, led_device, controller):
+        """Initialize the switch."""
+        self._prev_state = None
+        self._keypad_name = keypad_name
+        self._scene_name = scene_device.name
+        super().__init__(area_name, led_device, controller)
+
+    def turn_on(self, **kwargs):
+        """Turn the LED on."""
+        self._lutron_device.state = 1
+
+    def turn_off(self, **kwargs):
+        """Turn the LED off."""
+        self._lutron_device.state = 0
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attr = {}
+        attr["keypad"] = self._keypad_name
+        attr["scene"] = self._scene_name
+        attr["led"] = self._lutron_device.name
+        return attr
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._lutron_device.last_state
+
+    @property
+    def name(self):
+        """Return the name of the LED."""
+        return "{} {}: {} LED".format(
+            self._area_name, self._keypad_name, self._scene_name
+        )
+
+    def update(self):
+        """Call when forcing a refresh of the device."""
+        if self._prev_state is None:
+            self._prev_state = self._lutron_device.state
```

Comment:
Where is `self._prev_state` used?

Reply:
Seemingly nowhere..... I've removed it from the block of code I added.

Reply:
the name _prev_state was perhaps a misnomer. It's effectively a:
self._first_update..

Please don't submit it like this. You don't want to re-query the Main Repeater every time HASS wants to check the state. We get asynchronous notifications about switch state from the MR, which keeps the current state up to date.

So, what we need is a single update at startup, and that was the purpose of _prev_state. So rename it as you like (first_update bool, inital_State, whatever you like), but we should keep the explicit MR query to only happen at startup.

Reply:
@thecynic - Totally fair, but doesn't the underlying Lutron library actually cache the state ("last_state") ?

So, I guess what we really want to do is check last_state from Lutron and, if it's `None` do a force refresh? We don't want to add caching on top of caching, surely?

Reply:
@alistairg Ah, good point, that will work nicely. If 'last_state' is None, query 'state'. 👍

Reply:
Can't we use `add_entities(entities, True)` for this case? That will make sure that update is called during entity addition.

Reply:
I'm not sure honstly. Previously, I found update was called a lot, but maybe that's different now. It should be easy to test @alistairg if you want to double check that it only gets called once at startup.

Reply:
OK, so I updated the code that should now correctly address all scenarios. `update()` will trigger a low-level refresh from Lutron if there's no value (which is the case at first add) then will use the underlying cached value in future cases.